Modules/Loadable/Markups/MRML/vtkCurveGenerator.cxx

```diff
@@ -627,13 +628,34 @@ int vtkCurveGenerator::GeneratePointsFromFunction(vtkPoints* inputPoints, vtkPoi
   outputPedigreeIdArray->Reset();
   outputPedigreeIdArray->FillComponent(0, 0.0);
 
+  if (inputSurface)
+    {
+    this->SurfaceCellLocator->SetDataSet(inputSurface);
+    this->SurfaceCellLocator->BuildLocator();
+    }
+
   double previousPoint[3] = { 0.0 };
   for (int pointIndex = 0; pointIndex < totalNumberOfPoints; pointIndex++)
     {
     double sampleParameter = double(pointIndex) / ((double)(totalNumberOfPoints - 1));
     double curvePoint[3];
     this->ParametricFunction->Evaluate(&sampleParameter, curvePoint, nullptr);
-    outputPoints->InsertNextPoint(curvePoint);
+
+    if (inputSurface)
+      {
+      vtkIdType cellId = -1;
+      double closestCurvePointDist2 = std::numeric_limits<double>::max();
```

Comment:
```suggestion
      double closestCurvePointDist2 = VTK_DOUBLE_MAX;
```

In other classes making use `FindClosestPoint`, the closest point is initialized with `VTK_DOUBLE_MAX`. For sake of consistency, I suggest we keep the same convention.

https://github.com/Slicer/Slicer/blob/b4c6004ef2709bf71843b17a3364e2c0d22e455e/Modules/Loadable/Markups/VTKWidgets/vtkSlicerCurveRepresentation2D.cxx#L361-L371

Reply:
Is there any plan/desire to update from using `VTK_DOUBLE_MAX` in Slicer to the more standard and widely known `std::numeric_limits`?

Reply:
I am not aware of effort looking to remove these.
Looking at VTK master, it is still in used. See https://github.com/Kitware/VTK/search?q=VTK_DOUBLE_MAX

If you would like, you could discuss this on [VTK discourse](https://discourse.vtk.org). Cc'ing [Sean McBride](https://discourse.vtk.org/u/seanm/summary) may be sensible as they have been very [active](https://github.com/Kitware/VTK/commits?author=seanm) in modernizing the code base.

Reply:
Updated